File: lib/java/opentoken/src/main/java/com/truveta/opentoken/tokentransformer/NoOperationTokenTransformer.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.tokentransformer;

/**
 * A <code>No Operation</code> token transformer. No transformation is
 * applied whatsoever.
 */
public class NoOperationTokenTransformer implements TokenTransformer {

    /**
     * No operation token transformer.
     * <p>
     * Does not transform the token in any ways.
     */
    @Override
    public String transform(String token) {
        return token;
    }
}

File: lib/python/opentoken/src/main/opentoken/tokentransformer/no_operation_token_transformer.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from opentoken.tokentransformer.token_transformer import TokenTransformer


class NoOperationTokenTransformer(TokenTransformer):
    """
    A No Operation token transformer. No transformation is
    applied whatsoever.
    """

    def transform(self, token: str) -> str:
        """
        No operation token transformer.

        Does not transform the token in any ways.

        Args:
            token: The token to be transformed.

        Returns:
            The unchanged token.
        """
        return token
